File: leetcode/go/754.reach-a-number.go
func reachNumber(target int) int {
    if target < 0 {
        target = -target
    }
    i := 0
    x := 0
    for x != target {
        if x < target {
            i++
            x += i
        } else {
            diff := x - target
            if diff & 1 == 0 {
                break;
            } else {
                x = target - 1
            }
        }
    }
    return i
}

File: leetcode/rust-leetcode/src/754.reach-a-number.rs
impl Solution {
    pub fn reach_number(target: i32) -> i32 {
        let mut i = 0;
        let mut x = 0;
        let target = target.abs();
        while x != target {
            if x < target {
                i += 1;
                x += i;
            } else {
                let diff = (x - target);
                if diff & 1 == 0 {
                    x = target;
                    break;
                } else {
                    x = target - 1;
                }
            }
        }
        i
    }
}

